lib/spack/spack/platforms/linux.py

```diff
@@ -6,22 +6,22 @@
 import platform
 from spack.architecture import Platform, Target
 from spack.operating_systems.linux_distro import LinuxDistro
+import llnl.util.cpu as cpu
 
 
 class Linux(Platform):
     priority    = 90
 
     def __init__(self):
         super(Linux, self).__init__('linux')
-        self.add_target('x86_64', Target('x86_64'))
-        self.add_target('ppc64le', Target('ppc64le'))
 
-        self.default = platform.machine()
-        self.front_end = platform.machine()
-        self.back_end = platform.machine()
+        for name in cpu.supported_target_names():
+            self.add_target(name, Target(name))
 
-        if self.default not in self.targets:
-            self.add_target(self.default, Target(self.default))
+        # Get specific default
+        self.default = cpu.detect_host().name
+        self.front_end = self.default
+        self.back_end = self.default
```

Comment:
Same here -- can this be moved up into `Platform`?

Reply:
See above